### DISCUSSION TITLE
[v3] propertySynthesizers

### DISCUSSION THREAD

**Discussion by sakanjo** [livewire/livewire#6379] [1. Help]
```php
<?php

namespace App\Livewire;

use Livewire\Component;

class CreateSomething extends Component
{
    public string|int $value = 'Default';

    public function render()
    {
        return view('livewire.create-something');
    }
}
```

```html
<form>
    <input wire:model.live="value" />
    <p>{{ $value }}</p>
</form>
```

since primitive types aren't supported in [HandleComponents](https://github.com/livewire/livewire/blob/main/src/Mechanisms/HandleComponents/HandleComponents.php#L14) it's gonna coerce $value to be int which is `0`

only appears when using multiple-types property

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147
